hugr-core/src/extension/op_def.rs

```diff
@@ -494,6 +496,87 @@ mod test {
         const EXT_ID: ExtensionId = "MyExt";
     }
 
+    #[derive(serde::Serialize, serde::Deserialize, Debug)]
+    pub struct SimpleOpDef(OpDef);
+
+    impl SimpleOpDef {
+        pub fn new(op_def: OpDef) -> Self {
+            assert!(op_def.constant_folder.is_none());
```

Comment:
Nice! I was going to comment before, that `SimpleOpDef` was actually `OpDefEqualityWrapper` and the "simple" only came from the `impl Arbitrary for`, but this enforces simplicity, like it :)